src/weights.jl

```diff
@@ -291,3 +291,81 @@ end
 
 wmedian(v::RealVector, w::RealVector) = median(v, weights(w))
 wmedian{W<:Real}(v::RealVector, w::WeightVec{W}) = median(v, w)
+
+
+
+
+###### Weighted quantile #####
+# type 7 for now, as in base
+
+function Base.quantile{W<:Real}(v::RealVector, q::RealVector, w::WeightVec{W})
+
+    isempty(v) && error("median of an empty array is undefined")
+    isempty(q) && throw(ArgumentError("empty quantile array"))
+
+    # make sure the quantiles are in [0,1]
+    qpermute = sortperm(q)
+    q = q[qpermute]
+    q = bound_quantiles(q)
+
+    if w.sum == 0
+        error("weight vector cannot sum to zero")
+    end
+    if length(v) != length(w)
+        error("data and weight vectors must be the same size")
```

Comment:
Could you add `"got $length(v) and $length(w)"`? Always useful to provide more details.